### PR TITLE
Add board information tool

### PR DESCRIPTION
**Pull request by mschwan-phytec**
Add a tool that prints information about various parameters of the current hardware.

### CODE REVIEW

**Comment by mschwan-phytec** [comment]
This is work-in-progress.

@fshrOSB, @tbueOSB: Could you give some feedback on the output of the board info tool? Which exact parameters are needed? How should the output be formatted?

File: recipes-support/phytec-board-info/files/phytec-board-info.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		cat /etc/hostname
+		;;
+	-v|--machine-version)
+		# Hardware version is hard to find out. The only way I can think of is
+		# using the machine name's version
+		rev /etc/hostname | cut -f 1 -d '-' | rev
+		;;
+	-s|--serial)
+		# Serial number is tied to MAC address of the hardware module
+		cut -f 1- -d ':' --output-delimiter '' /sys/class/net/eth0/address
```
Comment (fshrOSB) [inline recipes-support/phytec-board-info/files/phytec-board-info.sh]: Somehow I am having difficulties when executing this option with the awsclient.
The awsclient only gets the first 2 characters when reading from stdout.
I suspect it has something to do with replacing the : with a null terminated empty string, but I am not sure what exactly is happening.
The following does work for me though.
`cat /sys/class/net/eth0/address | tr -d ':'`

Comment (mschwan-phytec) [reply recipes-support/phytec-board-info/files/phytec-board-info.sh]: Ok. Will replace this with `tr`, which actually looks nicer. :+1:

**Comment by mschwan-phytec** [comment]
The serial number option now prints the serial number defined by the device certificate.

File: recipes-support/phytec-board-info/files/phytec-board-info.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		cut -f 1 -d '' /proc/device-tree/compatible
+		;;
+	-m|--machine)
+		cat /etc/hostname
+		;;
+	-v|--machine-version)
+		cat /proc/device-tree/model | cut -d ' ' -f 2
+		;;
+	-s|--serial)
+		echo echo "0x$(openssl x509 -noout -serial -in /mnt/config/aws/certs/devcert.pem | cut -d '=' -f 2 | awk '{print tolower($0)}')"
```
Comment (motto-phytec) [inline recipes-support/phytec-board-info/files/phytec-board-info.sh]: please remove one echo

File: recipes-support/phytec-board-info/files/phytec-board-info.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+case "$1" in
+	-c|--compatible)
+		cut -f 1 -d '' /proc/device-tree/compatible
+		;;
+	-m|--machine)
+		cat /etc/hostname
+		;;
+	-v|--machine-version)
+		cat /proc/device-tree/model | cut -d ' ' -f 2
```
Comment (motto-phytec) [inline recipes-support/phytec-board-info/files/phytec-board-info.sh]: the machine version should be the hardware version or product version. At the moment the device itself do not know it.
it is ok for the first implementation.

**Review by motto-phytec — CHANGES_REQUESTED** [review]

File: recipes-support/phytec-board-info/files/phytec-board-info.sh (hunk 2 above)
Comment (mschwan-phytec) [reply recipes-support/phytec-board-info/files/phytec-board-info.sh]: Whoops, too much echo here indeed. :smile:

File: recipes-support/phytec-board-info/files/phytec-board-info.sh (hunk 3 above)
Comment (mschwan-phytec) [reply recipes-support/phytec-board-info/files/phytec-board-info.sh]: This is strange. Somehow an old version of this branch slipped into this. I remember having the machine version printed here. :thinking: I will change this to print the machine version...

**Pull request merged**